pytorch_toolkit/formula_recognition/im2latex/models/text_recognition_heads/attention_based.py

```diff
@@ -93,6 +100,14 @@ def forward(self, features, formulas=None):
         logits: [B, MAX_LEN, VOCAB_SIZE]
         """
         # encoding
+        if self.positional_encodings:
+            old_shape = features.shape
+            pe = PositionalEncodingPermute2D(channels=self.encoder_input_size)
+            encoded = pe(features)
+            encoded = encoded.expand(features.shape)
```

Comment:
Is this `expand` required? Does not standard broadcasting do the job automatically?

Reply:
Did not investigate this, will test.
Also, it is one of the implementation features of the Positional encoding, it returns encoding with batch dimension equal to one, though it is expected: positional encoding depends only on position, so for every image in batch with the same dimension it will be equal.

Reply:
I've tested, it works without `expand`.

Reply:
As expected.